extensions/amp-story/1.0/amp-story-interactive.js

```diff
@@ -381,6 +406,29 @@ export class AmpStoryInteractive extends AMP.BaseElement {
       true /** callToInitialize */
     );
 
+    // Check if the component page is active, and add class.
+    this.storeService_.subscribe(
+      StateProperty.CURRENT_PAGE_ID,
+      (currPageId) => {
+        if (currPageId === this.getPageId_()) {
+          this.timer_.delay(
+            () => {
+              this.mutateElement(() => {
+                this.rootEl_.classList.add(
+                  'i-amphtml-story-interactive-active'
+                );
+              });
+            },
+            this.storeService_.get(StateProperty.UI_STATE) ===
+              UIType.DESKTOP_PANELS
+              ? 350
+              : 0
+          );
```

Comment:
What's the benefit of setting a delay here rather than using css transition delays?
We could use `i-amphtml-story-desktop-panels` in our stylesheets to detect if `amp-story` is in desktop mode.

Reply:
Good question, there's no definitive answer. In your codepen you were using transition delays to offset the animations between the different options, which would be harder to use: `calc(0.2s + var(--i-amphtml-interactive-animation-delay)))` vs just `0.2s`. Same thing would apply for any animation delays. And transitions that won't need a delay (eg: the results page), you wouldn't need to get the variable and set it to the animation-delay, just add the animations/transitions.

Reply:
I like the idea of having all timing logic in one place. 
However I see your point that it's verbose CSS. 

I would default to keeping concerns separated. 
Since this timing has to do with style, it feels logical to let CSS control it.

Reply:
+1 to handle the delay from the CSS.
`i-amphtml-story-interactive-active` is true even as the desktop page is transitioning (videos are playing etc)

Reply:
Added commit that removes the delay here and adds a CSS variable for it